kvserver: unskip v2 flow control integration tests under duress

kvoli opened the pull request:
Similar to the prior commit, un-skip the v2 flow control integration
tests under duress.

These were disabled due to flakiness.

Resolves: https://github.com/cockroachdb/cockroach/issues/132272
Release note: None

kvoli committed:
kvserver: unskip v1 flow control integration tests under duress

These were skipped to unblock merging #132125 and later (presumed to
be) fixed by #132563.

Un-skip all `TestFlowControl.*` v1 integration tests under duress.

Resolves: #132310
Release note: None

kvoli committed:
kvserver: unskip v2 flow control integration tests under duress

Similar to the prior commit, un-skip the v2 flow control integration
tests under duress.

These were disabled due to flakiness.

Resolves: #132272
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132571)
<!-- Reviewable:end -->

kvoli committed:
testutils,kvserver: disable store rebalancer when replication manual

`ReplicationManual` can be specified in tests which use a test cluster
to prevent lease/replica movements.

This proves vital if the test is asserting exactly on the placement of a
range, or its leaseholder.

Also disable the store rebalancer, which could move both leases and
replicas under `ReplicationManual` before.

Informs: #132310
Informs: #132272
Release note: None

kvoli commented:
The extended failure:

```
TestFlowControlLeaderNotLeaseholder
[17:40:41 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/17313707?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildTestsSection=true&showLog=17313703_63242_63242&logFilter=debug&logView=flowAware)      No pass/skip/fail event found for test
[17:40:41 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/17313707?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildTestsSection=true&showLog=17313703_63243_63243&logFilter=debug&logView=flowAware)      === RUN   TestFlowControlLeaderNotLeaseholder
          test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestFlowControlLeaderNotLeaseholder1548164070
          test_log_scope.go:76: use -show-logs to present logs inline
      fatal error: too many concurrent timer firings
```

Is related to the go upgrade: https://github.com/cockroachdb/cockroach/issues/132573.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 3 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132571#-:-O9BZjtuE_mvZuKyE6JH:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli)



<!-- Sent from Reviewable.io -->

kvoli commented:
I'm going to merge this, since CI is reporting it as green -- there may still be some flaky behavior w.r.t slowness causing recently added replicas to form a send queue. We can address those issues separately.


TYFTR

bors r=sumeerbhola
